front/src/common/HelpModal/HelpModal.tsx

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect } from 'react';
+
+import { useTranslation } from 'react-i18next';
+import ReactMarkdown from 'react-markdown';
+import remarkGfm from 'remark-gfm';
+
+import ModalBodySNCF from 'common/BootstrapSNCF/ModalSNCF/ModalBodySNCF';
+import ModalHeaderSNCF from 'common/BootstrapSNCF/ModalSNCF/ModalHeaderSNCF';
+
+function HelpModal() {
+  const { t } = useTranslation('home/navbar');
+  const [helpModalText, setHelpModalText] = useState<string>('');
+
+  useEffect(() => {
+    const checkHelpMarkdown = async () => {
+      const expectedFileFirstLine = '<!-- OSRD Help Markdown File -->';
+      const defaultText = '# contact@osrd.fr';
+
+      try {
+        const response = await fetch('/help.md');
+
+        if (!response.ok) {
+          setHelpModalText(defaultText);
+          return;
+        }
+
+        const text = await response.text();
+        setHelpModalText(
+          text.startsWith(expectedFileFirstLine)
```

Comment:
Why do we discard the file if it doesn't start with `expectedFileFirstLine`?

Reply:
We always respond with the index file to every url that doesn't match a disk file in production (standard behavior for the path to be passed to the front router). I've detailed it in the PR text

Reply:
Oh my bad! Makes sense for a SPA.

Reply:
If we wanted to workaround this, we could either:

- Have a dedicated `/assets/` directory for static assets, the SPA index is never served there.
- Disallow paths ending with a file extension for SPA paths.
- Add `Accept: text/markdown` so that the server refuses to serve a non-Markdown file.

But these are larger changes.

Reply:
All these changes would require modifications of the gateway and they would need to create a service that would encapsulate the static_file service that is currently used.

I was not a real fan of modifying the gateway for this, it would make the PR way larger.

Reply:
The latter could maybe not require any gateway changes -- or we could even check the `Content-Type` returned by the server instead of having a magic header. Anyways, the current approach is fine by me as well.

Reply:
> The latter could maybe not require any gateway changes -- or we could even check the `Content-Type` returned by the server instead of having a magic header. Anyways, the current approach is fine by me as well.

That's true, I thought the Gateway might not implement this by default but it might be the case. I'm currently checking it.

Reply:
![image](https://github.com/user-attachments/assets/0f11d3f9-7381-4fe3-917e-6a4668314984)

Unfortunately, that's what I thought : we need to modify the gateway too if we want this behavior.

Reply:
Right, but would my suggestion in the previous comment work? Could we replace the `text.startsWith(expectedFileFirstLine)` check with `response.headers.get('Content-Type') === 'text/markdown'`?

Reply:
Oh like that, sorry, I thought you meant server side check. I guess that would work yes.

Reply:
Looks good, thank you!